### ISSUE TITLE
HomeGenie will not load after latest update

### ISSUE THREAD

**Issue by soxfan1966** [genielabs/HomeGenie#379]
I am running HomeGenie on a Raspberry pi via the X10Hub from tuicemen.  It has been running smoothly and I do periodic updates when they are reported.

This morning, I saw there was an update so I took the update.  However, since it finished updating I cannot get HomeGenie to load.

In working with Tuicemen on the x10 board:
http://forums.x10.com/index.php?topic=30955.0

He suspects it may have something to do with a recent password encryption change.

I can connect to the Pi via Putty and here is the info it reported:
![Capture2](https://user-images.githubusercontent.com/49078444/55259455-1415d000-523c-11e9-81ca-1d461b99a358.PNG)

Any help getting it up and running again would be appreciated

**Comment by tuicemen** [comment]
@soxfan1966  you may wish to look at issue #377 @genemars  posted about still being able to use basic authorization with a file edit. I'm not sure if this would work if you updated with out removing the basic Authorization prior. However you may have nothing to lose at this point.

**Comment by genemars** [comment]
@soxfan1966 as a possible work-around try a fresh install and then restore the configuration backup.

**Comment by soxfan1966** [comment]
How would I do a fresh install ?  I can access the Pi via Putty but thats about it.

**Comment by genemars** [comment]
@soxfan1966 I forgot to mention that the configuration backup must NOT have the password protection set, otherwise it will not work. 
You can uninstall old homegenie with a `apt remove homegenie` then follow the instruction on HG documentation site: https://genielabs.github.io/HomeGenie/#/get_started

**Comment by soxfan1966** [comment]
I have to assume my configuration backup (the zip file) has the password still associated as that was set when I backed it up.

**Comment by genemars** [comment]
So you will have to reconfigure HG, unless you decide to install an older release and repeat the update procedure following the instructions in the release notes.

**Comment by soxfan1966** [comment]
can I roll back the update bu installing the prior version and then using the saved configuration file ?

**Comment by soxfan1966** [comment]
I have a backup of the SD card image - thinking I will try using that and getting it onto an SD card and go from there.

**Comment by soxfan1966** [comment]
I was able to get the backed up image onto my PI's SD card and then use the saved backup file to restore.

Before I do, is there something I should do to remove my password for access before trying to update to the latest version of HG again?

**Comment by genemars** [comment]
@soxfan1966 if you were able to restore a password protected backup on an older release, then remove the password protection and do a new backup so that in case the update fails, you will be able to restore the new backup on a fresh install of the latest release. This is because the latest release use a different encryption mechanism. After succesfull restore you can set the password again.

**Comment by soxfan1966** [comment]
I think I removed the password and made another config backup.  Assuming this is where the password is Cleared then I should be all set.
![Capture](https://user-images.githubusercontent.com/49078444/55276811-728b8e80-52ce-11e9-942b-3e5e2fb97cfd.PNG)

**Comment by genemars** [comment]
@soxfan1966 correct.

**Comment by soxfan1966** [comment]
OK - back up and running with version 1.2 of the PiX10Hub and my devices (along with Alexa) working again.

I am going to hold off taking any HG updates for awhile and make sure everything is good before proceeding with that.

Thank you for your help.   NOTE: My password in HG is still cleared, will hold off on that for awhile too.

**Comment by tuicemen** [comment]
@soxfan1966  since you have this resolved perhaps you can close the issue.

**Comment by soxfan1966** [comment]
All set - Thank You

**Comment by soxfan1966** [comment]
Wanted to follow up this afternoon.

With no password still, I am getting the HG info about available updates.

I actually read the info provided (I know, I know) and saw the info about the password and need to remove before installing.

So I wanted to clarify that the problem I ran into is not the fault of HG - rather it was the dreaded user error (failure to read important information).

I was able to take and install the latest update with no issues.

Thanks
